controllers/inferenceservice_route.go

```diff
@@ -157,91 +172,115 @@ func (r *OpenshiftInferenceServiceReconciler) findSupportingRuntimeForISvc(ctx c
 // Reconcile will manage the creation, update and deletion of the route returned
 // by the newRoute function
 func (r *OpenshiftInferenceServiceReconciler) reconcileRoute(inferenceservice *inferenceservicev1.InferenceService,
-	ctx context.Context, newRoute func(service *inferenceservicev1.InferenceService, enableAuth bool) *routev1.Route) error {
+	ctx context.Context, newRoute func(service *inferenceservicev1.InferenceService, enableAuth bool, enableGrpc bool) *routev1.Route) error {
 	// Initialize logger format
 	log := r.Log.WithValues("inferenceservice", inferenceservice.Name, "namespace", inferenceservice.Namespace)
 
 	desiredServingRuntime := r.findSupportingRuntimeForISvc(ctx, log, inferenceservice)
 
-	enableAuth := true
-	if desiredServingRuntime.Annotations["enable-auth"] != "true" {
-		enableAuth = false
-	}
-	createRoute := true
-	if desiredServingRuntime.Annotations["enable-route"] != "true" {
-		createRoute = false
+	// optionsList allows for creation of both grpc and http/s routes as desired
+	var optionsList [2]string
+
+	createRoute, enableAuth, enableGrpc := false, false, false
+
+	// If routes need to be exposed, then enable both grpc and http/s
+	if desiredServingRuntime.Annotations["enable-route"] == "true" {
+		createRoute = true
+		enableGrpc = true
+		optionsList[1] = "grpc"
+
+		if desiredServingRuntime.Annotations["enable-auth"] == "true" {
+			enableAuth = true
+			optionsList[0] = "https"
+		} else {
+			optionsList[0] = "http"
+		}
 	}
 
-	// Generate the desired route
-	desiredRoute := newRoute(inferenceservice, enableAuth)
-
-	// Create the route if it does not already exist
-	foundRoute := &routev1.Route{}
-	justCreated := false
-	err := r.Get(ctx, types.NamespacedName{
-		Name:      desiredRoute.Name,
-		Namespace: inferenceservice.Namespace,
-	}, foundRoute)
-	if err != nil {
-		if !createRoute {
-			log.Info("Serving runtime does not have 'enable-route' annotation set to 'True'. Skipping route creation")
-			return nil
+	for _, v := range optionsList {
```

Comment:
I would prefer extracting the code inside this `for` to a function, so that you can call inline in the `if` block that is up these lines avoiding the need for this `for` loop.